packages/core/src/fhir-deduplication/deduplicate-fhir.ts

```diff
@@ -201,19 +235,110 @@ function replaceResourceReferences(
   return JSON.parse(updatedArrays);
 }
 
+type ResourceFilter = (entry: Resource, link: string) => Resource | undefined;
+
+const compositionFiltersMap = new Map<
+  string,
+  typeof removeResource | typeof removeDanglingReferences
+>([["all", removeDanglingReferences]]);
+
+const medicationRelatedFiltersMap = new Map<string, ResourceFilter>([
+  ["Medication", removeResource],
+]);
+
+const encounterFiltersMap = new Map<string, ResourceFilter>([
+  ["Condition", removeDanglingReferences],
+]);
+
+const diagReportFiltersMap = new Map<string, ResourceFilter>([
+  ["Observation", removeDanglingReferences],
+  ["Encounter", removeDanglingReferences],
+]);
+
+const resourceFiltersMap = new Map<string, Map<string, ResourceFilter>>([
+  ["DiagnosticReport", diagReportFiltersMap],
+  ["Encounter", encounterFiltersMap],
+  ["MedicationStatement", medicationRelatedFiltersMap],
+  ["MedicationRequest", medicationRelatedFiltersMap],
+  ["MedicationAdministration", medicationRelatedFiltersMap],
+  ["Composition", compositionFiltersMap],
```

Comment:
what about references to Practitioners, Locations, that will be dangling in all different resource types?

Reply:
Will add those! Locations was found to be a common issue with the FHIR validator

Reply:
added more dangling link handling logic